econml/inference/_inference.py

```diff
@@ -201,6 +201,41 @@ def effect_inference(self, X, *, T0, T1):
                                       feature_names=self._est.cate_feature_names(),
                                       output_names=self._est.cate_output_names())
 
+    def marginal_effect_inference(self, T, X):
+        X, T = self._est._expand_treatments(X, T, transform=False)
+
+        cme_inf = self.const_marginal_effect_inference(X)
+        if not self._est.treatment_featurizer:
+            return cme_inf
+
+        feat_T = self._est.transformer.transform(T)
```

Comment:
Maybe. I wanted to minimize changes to existing code. If I'm understanding your idea correctly, your approach would require me to change every line that calls _expand_treatments to process the output of the function differently (because I'll be returning X, T, and now additionally feat_T).